src/GeckoV2Migrator.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router02.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IERC20.sol";
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+import {AgentTokenV2} from "./AgentTokenV2.sol";
+import {AirdropClaim} from "./AirdropClaim.sol";
+
+
+/// @title GeckoV2Migrator
+/// @notice The following is contract to migrate the Gecko V1 token to V2
+/// The contract will:
+/// - Deploy the new Gecko V2 token contract
+/// - Deploy an airdrop contract for holders of V1 to claim the new token
+/// - Create and fund a liquidity pool on Uniswap
+/// - Distribute tokens to the AgentCoin DAO, Agent Wallet, and the pool
+contract GeckoV2Migrator is Ownable {
+    error AlreadyDeployed();
+    error NoEthToDeploy();
+    error NoTokensToDeploy();
+    error AlreadyMigrated();
+
+    event LiquidityPoolCreated(address pair);
+
+    IUniswapV2Router02 public immutable uniswapRouter;
+
+    uint256 public immutable agentAmount;
+    uint256 public immutable daoAmount;
+    uint256 public immutable airdropAmount;
+    uint256 public immutable poolAmount;
+    address public immutable agentcoinDao;
+    address public immutable agentWalletAddress;
+    string public name;
+    string public symbol;
+
+    address public immutable geckoV1;
+    address public geckoV2;
+    address public airdrop;
+
+    bool public hasMigrated;
+
+    constructor(
+        address owner, 
+        string memory _name, 
+        string memory _symbol, 
+        address _agentcoinDao,
+        address _agentWalletAddress, 
+        uint256 _daoAmount, 
+        uint256 _agentAmount, 
+        uint256 _airdropAmount, 
+        uint256 _poolAmount, 
+        address _geckoV1, 
+        address _uniswapRouter
+    ) Ownable(owner) {
+        name = _name;
+        symbol = _symbol;
+        agentcoinDao = _agentcoinDao;
+        agentWalletAddress = _agentWalletAddress;
+        daoAmount = _daoAmount;
+        agentAmount = _agentAmount;
+        airdropAmount = _airdropAmount;
+        poolAmount = _poolAmount;
+        geckoV1 = _geckoV1;
+        uniswapRouter = IUniswapV2Router02(_uniswapRouter);
+    }
+
+    /// @notice Migrate the tokens from Gecko V1 to V2
+    /// The contract will deploy the new Gecko V2 token, an airdrop contract for holders to claim the new token, and create a liquidity pool on Uniswap
+    /// It will also distribute tokens to the AgentCoin DAO, Agent Wallet, and the pool
+    /// @dev This function can only be called once
+    function migrate() external onlyOwner {
+        if (hasMigrated) {
+            revert AlreadyMigrated();
+        }
+
```

Comment:
Could add a check here to make sure the V1 contract has in fact been stopped, making sure we don't accidentally try to "migrate" before the stop function has been called.
```
if (!geckoV1.isStopped()) {
  revert V1NotStopped()
}
```